gazelle/python/testdata/binary_targets/BUILD.out

```diff
@@ -0,0 +1,35 @@
+load("@rules_python//python:defs.bzl", "py_binary", "py_library", "py_test")
+
+py_library(
+    name = "binary_targets",
+    srcs = [
+        "bar.py",
+        "baz.py",
+    ],
+    visibility = ["//:__subpackages__"],
+)
+
+py_binary(
+    name = "bar_binary",
+    srcs = ["bar_binary.py"],
+    visibility = ["//:__subpackages__"],
+    deps = [":binary_targets"],
+)
+
+py_binary(
+    name = "single_quote_main",
+    srcs = ["single_quote_main.py"],
+    visibility = ["//:__subpackages__"],
+    deps = [":bar_test"],
+)
```

Comment:
test targets that have `if __name__ == "__main__"` usually are for running them as `py_test`. e.g. the most common case is:
```
if __name__ == "__main__":
    unittest.main()
```

this code would break in such scenarios.

I think we should only create `py_binary` from regular files.

Reply:
Yeah, I adjusted the order of the if statements near the top of `generate.go`, so if a file ends with `_test.py` then we will always make it into a `py_test` target and we don't check for `if __name__ == "__main__"`. So I think doing this with unittest will still work as desired.

Reply:
I somehow made a mental jump to conclude that this was generating a pybinary for an ifmain that was in a test file, but it actually was just depending on a test file. I haven't seen cases where this would be needed, but if it does not make the code unnecesarily complex, I am fine with keeping it.